src/V3Sched.cpp

```diff
@@ -870,6 +915,9 @@ void createEval(AstNetlist* netlistp,  //
         textp->text(VString::replaceWord(textp->text(), "act", "nba"));
     });
 
+    // Commit trigger awaits from the previous time slot
+    if (timingKit.m_commitFuncp) funcp->addStmtsp(new AstCCall{flp, timingKit.m_commitFuncp});
+
```

Comment:
Are we sure this need not be in the loop?

I think we can both suspend and resume in the same time slot:

```
initial forever begin
  @(edge a);
  b = ~b;
end

initial forever begin
  @(edge b);
  a = ~a;
end
```

Reply:
Note: The only thing we don't want to is this to print both on the same iteration of 'act':

@(edge a);
$display("1");
@(edge a);
$display("2");

Different iterations within the same time slot could result in resuming both though.

Reply:
> Are we sure this need not be in the loop?

One detail is that `commit` also happens after each `resume`:
```
if (triggered) {
    trigsched.resume();
    trigsched.commit();
}
```
So the case with two `forever`s works.

I'm not sure where the global `commit` should happen yet; I need to come up with a good test for it. I think it should be in a loop, but it can't be in `act`, because that will cause self-triggering:
```
->e;  // triggered = 1
@e;  // next iter it gets resumed
```

Reply:
Isn't this example undefined by the standard?

Reply:
Agreed we need a test that breaks without that

Reply:
I think I have the right apporach now:
* if a trigger is not set, commit at the start of the iteration,
* if it is set, defer the commit until after resumption in the current iteration.
With this, there are no double triggers and no self-triggers.